### PR TITLE
Improve stash api

### PR DESCRIPTION
**Pull request by jamal-khey**
DELETE /network-modifications (onlyStashed=false) -> delete all modifications (also stashed modification)
DELETE /network-modifications?onlyStashed=true -> delete all stashed modifications
GET /network-modifications (onlyStashed=false) -> get all modifications (also stashed modifications)
GET /network-modifications?onlyStashed=true -> get all stashed modifications
PUT /network-modifications?stashed=true -> stash modification
PUT /network-modifications?stashed=false -> restore modification

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 1]
```diff
@@ -152,6 +152,25 @@ public ModificationInfos getModificationInfo(UUID modificationUuid) {
             .toModificationInfos();
     }
 
+    //get all modifications infos
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:155]: Use the same conventions for comment as elsewhere ?

**Comment by SlimaneAmar** [comment]
Remove code smells

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 2]
```diff
@@ -81,6 +81,33 @@ public ModificationInfos getNetworkModification(UUID networkModificationUuid) {
...
+        Map<UUID, ModificationInfos> modificationInfosMap = modifications.stream().collect(Collectors.toMap(ModificationInfos::getUuid, m -> m));
+        //put metadata in a hashmap
+        Map<UUID, ModificationInfos> metaDataMap = metaData.stream().collect(Collectors.toMap(ModificationInfos::getUuid, m -> m));
+
+        modificationInfosMap.forEach((uuid,modificationInfos) -> {
+            ModificationInfos metadata = metaDataMap.get(uuid);
+            if (modificationInfos != null) {
+                modificationInfos.setStashed(metadata.getStashed());
+                modificationInfos.setDate(metadata.getDate());
+            }
```
Comment (jamal-khey) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:106]: @SlimaneAmar  j'ai un doute sur cette implementation.
sans faire ca, l'attribut stashed est null, je me demande si il y a un autre moyen de recuperer ce status

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 3]
```diff
@@ -144,13 +144,35 @@ public ResponseEntity<ModificationInfos> getNetworkModification(
...
     @ApiResponse(responseCode = "200", description = "The network modifications were deleted")
     public ResponseEntity<Void> deleteNetworkModifications(
-            @Parameter(description = "Network modification UUIDs") @RequestParam("uuids") List<UUID> networkModificationUuids,
-            @Parameter(description = "Group UUID") @RequestParam("groupUuid") UUID groupUuid) {
-        networkModificationService.deleteNetworkModifications(groupUuid, networkModificationUuids);
+            @Parameter(description = "Network modification UUIDs") @RequestParam(name = "uuids", required = false) List<UUID> networkModificationUuids,
+            @Parameter(description = "Group UUID") @RequestParam("groupUuid") UUID groupUuid,
+            @Parameter(description = "delete only stashed modifications") @RequestParam(name = "onlyStashed", required = false, defaultValue = "false") Boolean onlyStashed) {
+
+        if (onlyStashed && (networkModificationUuids == null || networkModificationUuids.isEmpty())) {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: corrigé

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 4]
```diff
@@ -144,13 +144,35 @@ public ResponseEntity<ModificationInfos> getNetworkModification(
         return ResponseEntity.ok().body(networkModificationService.getNetworkModification(networkModificationUuid));
     }
 
+    @GetMapping(value = "/network-modifications", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get all network modification")
+    @ApiResponse(responseCode = "200", description = "The network modifications were returned")
+    public ResponseEntity<List<ModificationInfos>> getAllNetworkModification(
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: corrigé

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 5]
```diff
@@ -144,13 +144,35 @@ public ResponseEntity<ModificationInfos> getNetworkModification(
...
+            @Parameter(description = "Group UUID") @RequestParam("groupUuid") UUID groupUuid,
+            @Parameter(description = "get only stashed modifications") @RequestParam(name = "onlyStashed", required = false, defaultValue = "false") Boolean onlyStashed) {
+
+        if (!onlyStashed) {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: corrigé

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:106]: only in English !

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 6]
```diff
@@ -202,23 +208,18 @@ public ResponseEntity<UUID> createModificationInGroup(@RequestBody ModificationI
...
-
-    @PostMapping(value = "/network-modifications/restore", produces = MediaType.APPLICATION_JSON_VALUE)
-    @Operation(summary = "restore network modifications")
-    @ApiResponse(responseCode = "200", description = "The network modifications were restored")
-    public ResponseEntity<Void> restoreNetworkModifications(
-            @Parameter(description = "Network modification UUIDs") @RequestParam("uuids") List<UUID> networkModificationUuids,
-            @Parameter(description = "Group UUID") @RequestParam("groupUuid") UUID groupUuid) {
-        networkModificationService.restoreNetworkModifications(networkModificationUuids);
+            @Parameter(description = "Group UUID") @RequestParam("groupUuid") UUID groupUuid,
+            @Parameter(description = "stash or unstash network modifications") @RequestParam(name = "stashed", required = true) Boolean stashed) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:217]: Remove required (true is default value)

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 7]
```diff
@@ -47,12 +47,12 @@ public NetworkModificationController(NetworkModificationService networkModificat
...
                                                                            @Parameter(description = "Only metadata") @RequestParam(name = "onlyMetadata", required = false, defaultValue = "false") Boolean onlyMetadata,
-                                                                        @Parameter(description = "Stashed modifications") @RequestParam(name = "stashed", required = false, defaultValue = "false") Boolean stashed,
+                                                                        @Parameter(description = "Stashed modifications") @RequestParam(name = "onlyStashed", required = false, defaultValue = "false") Boolean stashed,
                                                                            @Parameter(description = "Return 404 if group is not found or an empty list") @RequestParam(name = "errorOnGroupNotFound", required = false, defaultValue = "true") Boolean errorOnGroupNotFound) {
         return ResponseEntity.ok().body(networkModificationService.getNetworkModifications(groupUuid, onlyMetadata, errorOnGroupNotFound, stashed));
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:57]: Rename the parameter `stashed` by `onlyStashed`

Comment (jamal-khey) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java (hunk 6 above)
Comment (jamal-khey) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: done

**Review by SlimaneAmar — APPROVED** [review]

**Review by etiennehomer — APPROVED** [review]

**Comment by etiennehomer** [comment]
Tests OK

**Pull request merged**